protocol/message.go

```diff
@@ -39,10 +57,39 @@ type DirectoryProof struct {
 	AP    *m.AuthenticationPath
 	STR   *m.SignedTreeRoot
 	TB    *m.TemporaryBinding `json:",omitempty"`
-	Error ErrorCode           `json:",omitempty"`
+	Error ErrorCode
+}
+
+type DirectoryProofInEpoch struct {
+	Type  int
+	AP    *m.AuthenticationPath
+	STR   []*m.SignedTreeRoot
+	Error ErrorCode
+}
+
+type DirectoryProofs struct {
```

Comment:
I have a question about the new `Limit` field. Could you please elaborate on what you mean by "control the response length"?